atomics/include/desul/atomics/CUDA_CC_7.hpp

```diff
@@ -0,0 +1,161 @@
+/* 
+Copyright (c) 2019, Lawrence Livermore National Security, LLC
+and DESUL project contributors. See the COPYRIGHT file for details.
+Source: https://github.com/desul/desul
+
+SPDX-License-Identifier: (BSD-3-Clause)
+*/
+#ifndef DESUL_ATOMICS_CUDA_CC_7_HPP_
+#define DESUL_ATOMICS_CUDA_CC_7_HPP_
+
+#include<type_traits>
+#include<desul/atomics/CUDA_ASM.hpp>
+
+#define DESUL_CUDA_INTEGRAL_OP_ATOMICS(MEMORY_ORDER, MEMORY_SCOPE)                 \
+  template <typename T>                                                           \
+  __device__									  \
+  typename std::enable_if<std::is_integral<T>::value, T>::type atomic_fetch_add(  \
+      T* const dest, T value, MEMORY_ORDER, MEMORY_SCOPE) {                       \
+    return Impl::__atomic_fetch_add_simt(dest, value, GCCMemoryOrder<MEMORY_ORDER>::value, MEMORY_SCOPE());  \
+  }                                                                               \
```

Comment:
While you are at it you could have another macro that takes `fetch_add` as argument and reuse below